app/generators/base/base_repository.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Base
+  class BaseRepository
```

Comment:
I have few questions to you:
1. How is your idea for using that kind of class?
2. Why do you preparing `base` module and `base` repositories inside generators directory? It means that repository is somehow related only to generators?
3. How you understood the usage of ActiveRecord model in terms of adapter pattern? Do you think that it's a good idea?
4. Why it's better than using model in application directly?

Reply:
@wmaciejak 1,2. You are right. I should have repository folder and there just "Base" file where I can store my method that just takes argument (capacity ) and is able to create a new cinemahall with given capacity. Because I wanted to store somewhere method that I can call when I want to create new hall.
3. I read  [this ](https://www.thegreatcodeadventure.com/rails-refactoring-part-i-the-adapter-pattern/) and [this](https://www.sitepoint.com/using-and-testing-the-adapter-design-pattern/) before coding and tried to 'adapt' my code to this.
4. Because I don't know how to do relation between seats, seats_reservations and reservation (as it is in our schema), I drew on a paper how I think it will work and this kind of solution